src/Rules/NoEnvCallsOutsideOfConfigRule.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Larastan\Larastan\Rules;
+
+use Larastan\Larastan\Concerns\HasContainer;
+use PhpParser\Node;
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Rules\Rule;
+use PHPStan\Rules\RuleError;
+use PHPStan\Rules\RuleErrorBuilder;
+
+/**
+ * Catches `env()` calls outside of the config directory.
+ *
+ * @implements Rule<FuncCall>
+ */
+class NoEnvCallsOutsideOfConfigRule implements Rule
+{
+    use HasContainer;
+
+    public function getNodeType(): string
+    {
+        return FuncCall::class;
+    }
+
+    /** @return array<int, RuleError> */
+    public function processNode(Node $node, Scope $scope): array
+    {
+        $name = $node->name;
+
+        if ($name instanceof Expr) {
+            return [];
+        }
+
+        if ($name->toString() !== 'env') {
```

Comment:
is this check adequate if an `use function \Foo\env;` construct is used ?

Reply:
good catch! it did not